### PR TITLE
[rllib] Merge DDPG and DDPG2 implementations

### PR DESCRIPTION
**Pull request by andrewztan**
## What do these changes do?

Looked into implementation and performance differences between DDPG (https://github.com/ray-project/ray/tree/master/python/ray/rllib/ddpg) and DDPG2 (https://github.com/ray-project/ray/tree/master/python/ray/rllib/ddpg2) implementations. Richard and I decided that DDPG was a better and more thorough implementation. I removed DDPG2 and any mentions of DDPG2 in the codebase. I also added the tests used for the comparison so that the results in #2181 can be reproduced.

## Related issue number

#2181

### CODE REVIEW

**Comment by andrewztan** [comment]
@richardliaw

File: python/ray/rllib/tuned_examples/ddpg_vs_ddpg2_tests/pendulum-ddpg2.yaml
Comment (ericl) [inline python/ray/rllib/tuned_examples/ddpg_vs_ddpg2_tests/pendulum-ddpg2.yaml]: Let's drop the parent directory from all these files?

File: python/ray/rllib/tuned_examples/ddpg_vs_ddpg2_tests/pendulum-ddpg.yaml
Comment (ericl) [inline python/ray/rllib/tuned_examples/ddpg_vs_ddpg2_tests/pendulum-ddpg.yaml]: Could you add these notes for all the yaml files?

File: python/ray/rllib/tuned_examples/ddpg_vs_ddpg2_tests/pendulum-ddpg2.yaml
Comment (andrewztan) [reply python/ray/rllib/tuned_examples/ddpg_vs_ddpg2_tests/pendulum-ddpg2.yaml]: Yeah I can do that. The only thing is there already exists "montaincarcontinuous-ddpg.yaml" and "pendulum.ddpg" in /tuned_examples. Should I just replace those? 

The purpose of the yaml files in /ddpg_vs_ddpg2_tests was to compare the 2 implementations by using the (almost) same hyperparameters, not necessarily to find the configuration that performs best, which is why I decided to separate it from /tuned_examples. Also, should I remove the yaml files related to ddpg2? I left them just for comparison sake. Please let me know how you think we should proceed.

**Comment by ericl** [comment]
Yeah, let's get rid of ddpg2 files. The yamls should reflect the current
state of the repo.

File: python/ray/rllib/tuned_examples/ddpg_vs_ddpg2_tests/pendulum-ddpg2.yaml
Comment (richardliaw) [reply python/ray/rllib/tuned_examples/ddpg_vs_ddpg2_tests/pendulum-ddpg2.yaml]: Let's take out all the DDPG2 files. What are the performance differences for the current mountaincarcontinuous and pendulum yaml files compared to the new ones?

**Comment by richardliaw** [comment]
Nice point on keeping the comparison files. I think the fact that they're logged in #2181 is already enough, and as Eric mentioned, these example configurations should be in sync with the repository.

Let's make sure we have 1 set of DDPG files (as commented above), and we can merge right after.

File: python/ray/rllib/tuned_examples/ddpg_vs_ddpg2_tests/pendulum-ddpg.yaml
Comment (ericl) [inline python/ray/rllib/tuned_examples/ddpg_vs_ddpg2_tests/pendulum-ddpg.yaml]: Put these directly under `tuned_examples`?

**Review by richardliaw — APPROVED** [review]
Will merge after tests pass.

**Comment by richardliaw** [comment]
jenkins retest this please

**Comment by richardliaw** [comment]
jenkins retest this please

**Comment by richardliaw** [comment]
jenkins retest this please

**Comment by richardliaw** [comment]
jenkins retest this please

**Comment by richardliaw** [comment]
Test failures look unrelated. Thanks for putting this together @andrewztan!

**Pull request merged**
